example-get-started-experiments/code/.github/workflows/deploy-model-template.yml

```diff
@@ -0,0 +1,67 @@
+name: Test Model Artifact
+
+on:
+  # the workflow is triggered whenever a tag is pushed to the repository
+  push:
+    tags:
+      - "*"
+  # we can also run the workflow manually...
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Manual version name'
+        required: true
+        type: string
+  
+permissions:
+  contents: write
+  id-token: write
+
+jobs:
+
+  # This job parses the git tag with the GTO GitHub Action to identify model registry actions
+  parse:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: "Parse GTO tag"
+      id: gto
+      uses: iterative/gto-action@v2
+    outputs:
+      event: ${{ steps.gto.outputs.event }}
+      name: ${{ steps.gto.outputs.name }}
+      stage: ${{ steps.gto.outputs.stage }}
+      version: ${{ steps.gto.outputs.version }}
+
+  test-model-artifact:
+    needs: parse
+    # using the outputs from the "parse" job, we run this job only for actions
+    # in the model registry and only when the model was assigned to a stage called "test"
+    if: "${{ needs.parse.outputs.event == 'assignment' }} && ${{ needs.parse.outputs.stage == 'test' }}"
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.8'
+        cache: 'pip'
+        cache-dependency-path: requirements.txt
+
+    - run: pip install -r requirements.txt
+
+    # this step adds our project's DVC remote to the workflow environment
+    - name: Add DVC Remote
+      run:  dvc remote add -d --local storage s3://dvc-public/remote/get-started-pools
+    
+    # this step uses DVC to download the model from our remote repository and deploys the model
+    # Model deployment is mocked here as it is specific to each deployment environment
+    - name: Get Model and Check Size
+      run: |
+        dvc get model/model.pth # this will download the right model artifact from the DVC remote
```

Comment:
Let's use `dvc artifacts get` here (we can get all the args from the parse step).

Reply:
Sure, we can use it here, but we won't be able to use it with the sagemaker deployment - that one needs an S3 URL (that is specific to Sagemaker). I would suggest actually adding the --show-url functionality to dvc artifacts.

Reply:
That's a good idea. Even without that, though, I think we can get the url inside the sagemaker python script with only the gto info like:

```python
repo = 'https://github.com/iterative/example-get-started-experiments.git'
artifact = dvc.api.artifacts_show(
    name,
    version=version,
    stage=stage,
    repo=repo,
)
model_data = 
    artifact['path'],
    rev=artifact['rev'],
    repo=repo,
)
```

`dvc.api.artifacts_show` wasn't available when this was introduced, but now it should help here. Then we can drop this:

 https://github.com/iterative/example-repos-dev/blob/2f7a36d237e88b39083edcfc7e06f2c4fab5f3c9/example-get-started-experiments/code/.github/workflows/deploy-model.yml#L63

Reply:
@efiop The github action also adds a local s3 remote to override the http remote:

https://github.com/iterative/example-repos-dev/blob/2f7a36d237e88b39083edcfc7e06f2c4fab5f3c9/example-get-started-experiments/code/.github/workflows/deploy-model.yml#L60

Can we add the `config` option to `get_url` so we can add this info in the python call instead?

Reply:
By the way, no strong opinion that we do it this way for the sagemaker example. We can keep as is, use `dvc artifacts get --show-url`, or use the DVC API like I show above. Just thinking through the possible options.

Reply:
Personally, I would keep it like it is for now and replace it with `dvc artifacts get` later if we add the `--show-url` option. The sagemaker pipeline is not an integral part of the getting-stared guide, it is just another example we link to.

But the dvc.api approach is not visible form the workflow itself so it would require extra explanation.

Either way, I'd say it is not all that important what we do with the sagemaker workflow right now.